crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs

```diff
@@ -244,7 +268,8 @@ pub(crate) fn unused_import(checker: &Checker, scope: &Scope, diagnostics: &mut
     }
 
     let in_init = checker.path().ends_with("__init__.py");
-    let fix_init = checker.settings.preview.is_enabled();
+    let fix_init = !checker.settings.ignore_init_module_imports;
+    let preview_mode = checker.settings.preview.is_enabled();
```

Comment:
These three lines look fishy because I'd originally replaced this original code
```rust
    let fix_init = !checker.settings.ignore_init_module_imports;
```
with
```
    let fix_init = checker.settings.preview.is_enabled();
```
which was perhaps lazy. In this PR I'm restoring the old behavior, and so gave `checker.settings.preview.is_enabled();` its own name.